### ISSUE TITLE
Upstart init.d services supports respawn; can this be safely utilized in init.d scripting?

### ISSUE THREAD

**Issue by jeffaco** [Microsoft/omi#267]
Upstart init.d services supports respawn; can this be safely utilized in init.d scripting while not affecting SystemV init.d?

SystemV init.d (Redhat 5, Debian 6, Ubuntu 9) does not support respawn.
Upstart init.d (Redhat 6, Ubuntu 9-14) does support respawn
Systemctl is totally different, and does support respawn

Focusing on the first two: Is it possible to use feature-based detection to determine if respawn can be used to restart a failed omiserver process? If so, then postinstall scripts should be modified to support upstart-based systems to respawn omiserver on failure.

**Comment by JumpingYang001** [comment]
System V/Upstart/Systemd supported OS list:
```
System V is the older init system: 
Debian 6 and earlier
Ubuntu 9.04 and earlier
Redhat 5 and earlier
CentOS 5 and earlier

Upstart: 
Ubuntu 9.10 to Ubuntu 14.10, including Ubuntu 14.04
Redhat 6
CentOS 6

Systemd is the init system for the most recent distributions featured here:
Debian 7 and Debian 8
Ubuntu 15.04 and newer
Redhat 7
CentOS 7
```

Determine which way:
```
/usr/lib/systemd tells you you're on a systemd based system.
/usr/share/upstart is a pretty good indicator that you're on an Upstart-based system.
/etc/init.d tells you the box has SysV init in its history.
```
More info can refer from: http://unix.stackexchange.com/questions/196166/how-to-find-out-if-a-system-uses-sysv-upstart-or-systemd-initsystem

**Comment by yakman2020** [comment]
Implemented in commit 18c22df
